instructor/common.py

```diff
@@ -0,0 +1,604 @@
+                              context_instance=RequestContext(request))
+
+
+def western_blot_antibody_edit(request, assignment):
+    a = models.Assignment.objects.get(id=assignment)
+    (wb, created) = models.WesternBlot.objects.get_or_create(assignment=a)
+    wb.save()
+    WesternBlotAntibodyFormset = modelformset_factory(models.WesternBlotAntibody, extra=1, can_delete=True,
+                                                      exclude=['western_blot'])
+    message = ''
+    if request.method == "POST":
+        formset = WesternBlotAntibodyFormset(request.POST)
+        formset.clean()
+        if ( formset.is_valid()):
+            message = "Thank you"
+            entries = formset.save(commit=False)
+            for form in entries:
+                form.western_blot = wb
+                form.save()
+        else:
+            message = "Something went wrong"
+    return render_to_response('instructor/generic_formset.html',
+                              {'formset': WesternBlotAntibodyFormset(
+                                  queryset=models.WesternBlotAntibody.objects.filter(western_blot=wb)),
+                               'message': message,
+                               'assignment': a
+                              },
+                              context_instance=RequestContext(request))
+
+
+def western_blot_antibody_band_edit(request, assignment, antibody, sp):
+    a = models.Assignment.objects.get(id=assignment)
+
+    (wb, created) = models.WesternBlot.objects.get_or_create(assignment=a)
+    wb.save()
+
+    (protocol, created) = models.StrainProtocol.objects.get_or_create(id=sp)
+    protocol.save()
+
+    ab = models.WesternBlotAntibody.objects.get(id=antibody)
+    WesternBlotAntibodyBandFormset = modelformset_factory(models.WesternBlotAntibodyBands, extra=1, can_delete=True,
+                                                          exclude=['antibody', 'strain_protocol'])
+    message = ''
+    if request.method == "POST":
+        formset = WesternBlotAntibodyBandFormset(request.POST)
+        formset.clean()
+        if ( formset.is_valid()):
+            message = "Thank you"
+            entries = formset.save(commit=False)
+            for form in entries:
+                form.antibody = ab
+                form.strain_protocol = protocol
+                form.save()
+        else:
+            message = "Something went wrong"
+    return render_to_response('instructor/generic_formset.html',
+                              {'formset': WesternBlotAntibodyBandFormset(
+                                  queryset=models.WesternBlotAntibodyBands.objects.filter(antibody=ab,
+                                                                                          strain_protocol=protocol)),
+                               'message': message,
+                               'assignment': a
+                              },
+                              context_instance=RequestContext(request))
+
+
```

Comment:
A more general comment, this seems like a ton of copy/paste code and much of this could be simplified with making smarter form classes I think